src/client/mod.rs

```diff
@@ -143,6 +143,58 @@ impl<T: Transport, L: LoginCredentials> TwitchIRCClient<T, L> {
             .await
     }
 
+    /// Ban a user with a optional reason in the given Twitch channel.
+    pub async fn ban(
+        &self,
+        channel_login: String,
+        user: String,
+        reason: Option<String>,
+    ) -> Result<(), Error<T, L>> {
+        self.privmsg(
+            channel_login,
+            format!("/ban {} {}", user, reason.unwrap_or_default()),
+        )
+        .await
+    }
+
+    /// Unban a user in the given Twitch channel.
+    pub async fn unban(&self, channel_login: String, user: String) -> Result<(), Error<T, L>> {
+        self.privmsg(channel_login, format!("/unban {}", user))
+            .await
+    }
+
+    /// Timeout a user with a optional reason in the give Twitch channel.
```

Comment:
Same thing here, needs a segment to say that application-level errors are not detected (like above).